packages/mobile/src/redux/store.ts

```diff
@@ -1,21 +1,51 @@
 import AsyncStorage from '@react-native-community/async-storage'
 import { applyMiddleware, compose, createStore } from 'redux'
-import { createMigrate, persistReducer, persistStore } from 'redux-persist'
+import { createMigrate, getStoredState, persistReducer, persistStore } from 'redux-persist'
+import FSStorage from 'redux-persist-fs-storage'
 import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
 import createSagaMiddleware from 'redux-saga'
+import { PerformanceEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { APP_NAME } from 'src/config'
 import { migrations } from 'src/redux/migrations'
 import rootReducer from 'src/redux/reducers'
 import { rootSaga } from 'src/redux/sagas'
+import { ONE_DAY_IN_MILLIS } from 'src/utils/time'
+
+const timeBetweenStoreSizeEvents = ONE_DAY_IN_MILLIS
+let lastEventTime = Date.now()
 
 const persistConfig: any = {
   key: 'root',
   version: 7, // default is -1, increment as we make migrations
-  storage: AsyncStorage,
+  keyPrefix: `${APP_NAME}-`, // the redux-persist default is `persist:` which doesn't work with some file systems.
+  storage: FSStorage(),
   blacklist: ['home', 'geth', 'networkInfo', 'alert', 'fees', 'recipients', 'imports'],
   stateReconciler: autoMergeLevel2,
   migrate: createMigrate(migrations, { debug: true }),
+  serialize: (data: any) => {
+    // We're using this to send the size of the store to analytics while using the default implementation of JSON.stringify.
+    const stringifiedData = JSON.stringify(data)
+    // if data._persist or any other key is present the whole state is present (the content of the keys are
+    // sometimes serialized independently).
+    if (data._persist && lastEventTime + timeBetweenStoreSizeEvents > Date.now()) {
+      lastEventTime = Date.now()
+      ValoraAnalytics.track(PerformanceEvents.redux_store_size, {
+        size: stringifiedData.length,
+      })
+    }
+    return stringifiedData
+  },
 }
 
+const fetchAsyncStorageState = (config: any) =>
+  getStoredState({ ...config, storage: AsyncStorage, keyPrefix: 'persist:' })
+
+persistConfig.getStoredState = (config: any) =>
+  getStoredState(config)
+    .then((state) => state ?? fetchAsyncStorageState(config))
+    .catch(() => fetchAsyncStorageState(config))
```

Comment:
We should not swallow this error silently. Let's log it.
Are we sure we want to fallback on the AsyncStorage state in case an error is raised? 
The use case I'm concerned about is if an error is raised much later after the fs state has been used for a long time and probably can't be replaced with the old AsyncStorage state, as it could be surprising to the user, or even not match the current account anymore.

Reply:
Yeah, that makes sense. It should never ever happen, if it does it means we lost the state, which would be really bad. I'm not sure what's worse, if showing old (maybe very old) state or just losing everything. Definitely need to log this and I'll add a new analytics event to track this as well

Reply:
I ended up sending the error to Sentry instead of an analytics event, since it felt like a better place, but I'm open to changing it if you feel otherwise.